service-deployment/bootstrap/hadoop-service/hadoop-configuration/hdfs-start.sh.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import os
+import sys
+import kazoo
+import subprocess
+
+
+
+def execute_shell_with_output(shell_cmd, error_msg):
+
+    return_code = 0
+
+    try:
+        res = subprocess.check_output( shell_cmd, shell=True )
+
+    except subprocess.CalledProcessError as err:
+        print error_msg
+        return_code = err.returncode
+        res = err.output
+
+    return res, return_code
+
+
+
+def execute_shell(shell_cmd, error_msg):
+
+    try:
+        subprocess.check_call( shell_cmd, shell=True )
+
+    except subprocess.CalledProcessError:
+        print error_msg
+        sys.exit(1)
+
+
+
+if __name__ == "__main__":
+
+    # Will be filled when namenode bootstraping
+    host = {{ host_config['ip'] }}
```

Comment:
>host = {{ host_config['ip'] }} [](start = 4, length = 30)

how is this py file used? need an argument?

Reply:
template